src/realm/query_engine.cpp

```diff
@@ -98,41 +98,54 @@ size_t ParentNode::aggregate_local(QueryStateBase* st, size_t start, size_t end,
 
     m_state = st;
     m_state->set_payload_column(source_column);
-    size_t local_matches = 0;
 
     if (m_children.size() == 1) {
         return find_all_local(start, end);
     }
+    
+    if(m_children.size()>1)
+    {
+        const auto compressed = std::all_of(m_children.begin(), m_children.end(), [](const auto child){
```

Comment:
The Between Node is not copying with this case, @jedelbo if I construct a query like this: 
```

TableView results = Base::m_table->where()
                                        .greater(Base::m_col, Base::needles[i - 1].template get<underlying_type>())
                                        .less(Base::m_col, Base::needles[i].template get<underlying_type>())
                                        .find_all();
```

We first add the greater condition and later the less condition, then when we run find_all, it will basically process these conditions as children of the same ParentNode. I am not sure if this is the right way to handle this scenario, but the idea is that, if the children nodes are IntegerNodes and the leaf array is compressed, then we need to extract all the values in the advance and process them.